### PR TITLE
Account profile management 

### PR DESCRIPTION
**Pull request by jonas-atmire**
This PR is related to the rest contract update mentioned in: https://github.com/DSpace/Rest7Contract/pull/113

* Admin users can edit both the non-metadata information (email, netid etc) as well as the metadata information of all users
* Non admin user can edit their own metadata information
* Non admin user can not edit their non-metadata information (email, netid etc)
* Non admin users can not edit another person's information

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java [hunk 1]
```diff
@@ -44,6 +45,9 @@
     @Autowired
     AuthorizeService authorizeService;
 
+    @Autowired
+    DSpaceObjectMetadataPatchUtils metadataPatchUtils;
```
Comment (tdonohue) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java:49]: It's unclear why this is now `@Autowired` and imported into this class. I don't see any new code that uses it?  Are these changes to `EPersonRestRepository` necessary?

**Review by tdonohue — COMMENTED** [review]
@jonas-atmire : Overall, this PR looks fine to me. However, I didn't notice some minor things in the ITs, and I'm not sure I understand the changes to the `EPersonRestRepository` itself, and the newly imported util class seems unused? (maybe I'm overlooking something though?)

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java (hunk 1 above)
Comment (jonas-atmire) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java:49]: You are correct @tdonohue, this should not be present in this part of the code (Seems to be a remnant of an older commit that has been overlooked)
I'll get these feedback-points looked at. Thanks for the review

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java [hunk 2]
```diff
@@ -1516,6 +1517,102 @@ private void runPatchMetadataTests(EPerson asUser, int expectedStatus) throws Ex
...
+
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.metadata", Matchers.allOf(
+                            MetadataMatcher.matchMetadata("eperson.firstname", newName))));
+
+    }
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java:1578]: please use an additional get to check that changes are persisted

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java [hunk 3]
```diff
@@ -1516,6 +1517,102 @@ private void runPatchMetadataTests(EPerson asUser, int expectedStatus) throws Ex
...
+
+        String token = getAuthToken(ePerson2.getEmail(), password);
+
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isForbidden());
+
+    }
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java:1614]: please verify that no changes have been applied

**Review by abollini — COMMENTED** [review]
Hi @jonas-atmire 
thanks for this PR, it looks mostly ok to me, I have added few inline comments to improve the ITs

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public EPersonRestRepository(EPersonService dsoService) {
 
     @Override
     protected EPersonRest createAndReturn(Context context)
-            throws AuthorizeException {
+        throws AuthorizeException {
```
Comment (abollini) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java:57]: there are only formatting change in this class, can you revert them please?

**Review by abollini — COMMENTED** [review]
my feedback have been processed but unfortunately the latest commit have introduced a lot of unnecessary changes in the IT class due to formatting... I would like to have them reverted as otherwise they will cause conflict and more effort in coming PR about IT

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java (hunk 4 above)
Comment (tdonohue) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java:57]: I agree these seem like non-standard formatting changes. It'd be best to just revert all changes to `EPersonRestRepository`.  Could that be done quickly @jonas-atmire  or @MarieVerdonck ?

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now. All feedback has been addressed. Thanks @jonas-atmire and @MarieVerdonck

**Review by abollini — APPROVED** [review]
thanks for the cleanup, it looks better now!

**Pull request merged**